### PR TITLE
feat: default to all input files for affected tasks

### PR DESCRIPTION
**Pull request by affanshahid**
Just sending a PR for what I had to do locally to "fix" https://github.com/moonrepo/moon/issues/1329. Feel free to close this if someone else is already working on this or if my approach isn't ideal.

Wasn't sure where to put the "walking" code, one option was to create a fake FileGroup to re-use the logic from there but that felt hacky, so added a method to Task.

### CODE REVIEW

File: nextgen/task/src/task.rs [hunk 1]
```diff
@@ -159,6 +166,43 @@ impl Task {
...
+    pub fn get_input_files<S: AsRef<str>>(
+        &self,
+        workspace_root: &Path,
+        project_source: S,
+    ) -> miette::Result<Vec<ProjectRelativePathBuf>> {
+        let mut list = vec![];
+        let project_source = project_source.as_ref();
+
+        for path in &self.input_files {
+            list.push(path.strip_prefix(project_source).unwrap().to_owned());
```
Comment (milesj) [inline nextgen/task/src/task.rs:180]: I think this should return the paths relative from the workspace root.

The "relative from the project root" is only required for the affected logic, so I would move that part over there.

**Comment by milesj** [comment]
New features also go into the develop branch, can you change the base to `develop-1.22`.

**Comment by affanshahid** [comment]
I have made the requested changes

**Comment by milesj** [comment]
@affanshahid Implementation looks good, just needs some tests.

**Comment by affanshahid** [comment]
@milesj I've added tests, lmk if something seems off. However one thing I realised is that if the the set of input files is empty then the underlying command will be passed no arguments. This might break things for people expected a `.` in those scenarios, what do you think? Maybe add a "." arg as a fallback to be deprecated in a future breaking release?

**Comment by milesj** [comment]
@affanshahid Good point. Maybe in `runner.rs`, if files list is still empty, default to `.`.

**Comment by affanshahid** [comment]
@milesj I've added the fallback along with associated tests and a deprecation warning

**Comment by milesj** [comment]
@affanshahid changes looked good, just need to rebase the develop branch since changes landed there

**Comment by affanshahid** [comment]
@milesj Just did, also changed tests for deno, node, bun, system which expected a dot when no files are expected

**Comment by affanshahid** [comment]
@milesj I've rebased and updated the hashses

**Comment by milesj** [comment]
@affanshahid Awesome, thanks you!

**Pull request merged**